### PR TITLE
Add docker-style file variable support (fixes #189)

### LINKED ISSUE #189: [RFC] Add support for docker-style _FILE environment variables
I'd like to add support for docker-style `_FILE` environment variables. What this would mean basically is the following:

Take this line as an example:
```
DATABASES = {'default': env.db(default=sqlite_default)}
```
This will make `environ` look for a `DATABASE_URL` and use that. I'd like to extend it to first look for `DATABASE_URL_FILE` and if that variable exists and is set read the file it points to and use that instead of `DATABASE_URL`.

The idea here is that stuff like docker swarm secrets are injected as files into your container and I'd need an easy way to reference those. You can find more information about it at https://hub.docker.com/r/_/postgres/ (scroll down to `Docker Secrets`). Note that while I am referring solely to docker here, the same holds true for any other system which injects secrets as files and not environment variables.

Backwards compatibility wise there should be no issue;
```
env('ASD_FILE')
```
would try `ASD_FILE_FILE` and not read `ASD_FILE` as file directly...

If you think this is a good idea I'll happily supply a PR; if you are need more convincing please also tell me :)

### PR DESCRIPTION
**Pull request by SmileyChris**
Docker (swarm) and Kubernetes are two widely used platforms that store their Secrets in tmpfs inside containers as individual files.

These are a secure way to be able to share configuration data between containers, and having a way to access these in django-environ is useful.

Here's a quote from the official Postgres docker hub image:

> As an alternative to passing sensitive information via environment variables, `_FILE` may be appended to some of the previously listed environment variables, causing the initialization script to load the values for those variables from files present in the container. In particular, this can be used to load passwords from Docker secrets stored in `/run/secrets/<secret_name>` files. For example:

### CODE REVIEW

**Comment by sergeyklay** [comment]
I'll take a look ASAP

**Comment by SmileyChris** [comment]
Oh, we're supporting 3.5 still. Easy changes to make for that.
Windows CI fails because it can't read temp files 😒

**Comment by SmileyChris** [comment]
One design decision that should be considered: currently if you use a `_FILE` key on a missing file (or not enough permissions), then it will fail hard with an os exception -- the thinking being that failing loud is better than silently if you explicitly asked to use a file.

**Comment by sergeyklay** [comment]
@SmileyChris Could you please update the PR descrption by adding a bit more info about purpose and possible use cases.

**Comment by SmileyChris** [comment]
@sergeyklay updated description, pushed py3.5 fix, 100% coverage, and hopefully fixed windows ci tests too

**Comment by SmileyChris** [comment]
Lint fixed (pfft, should use black's 88 line length now :D)

**Review by sergeyklay — CHANGES_REQUESTED** [review]
In general, everything looks quite good, except for a few nitpicking from me.

File: README.rst [hunk 1]
```diff
@@ -88,6 +88,20 @@ environment variables obtained from an environment file and provided by the OS:
...
+.. code-block:: python
+
+    import environ
+
+    env = environ.FileAwareEnv()
+
+    # If a ``SECRET_KEY_FILE`` environment variable exists, its contents will be
+    # read from the file system and used instead of the ``SECRET_KEY``
+    # environment variable.
+    SECRET_KEY = env('SECRET_KEY')
```
[the remark this answers is not in the document]
Comment (SmileyChris) [reply README.rst:103]: Keeping it simple is good, but then I don't think anyone will find the feature if it lives there... Could I trim it down instead?

[the remark this answers is not in the document]
Comment (SmileyChris) [reply README.rst]: Ok, removed it (but added a short mention of how to do it in the "Features" bullet point that remains in README)

File: environ/environ.py [hunk 2]
```diff
@@ -786,6 +787,10 @@ def read_env(cls, env_file=None, **overrides):
             cls.ENVIRON.setdefault(key, value)
 
 
+class FileAwareEnv(Env):
+    ENVIRON = FileAwareMapping()
```
[the remark this answers is not in the document]
Comment (SmileyChris) [reply environ/environ.py:805]: Done

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Merged! Thank you for the patch, and for helping us make django-environ better.

**Comment by panteparak** [comment]
is there a timeline for the next release that includes this PR?

**Comment by sergeyklay** [comment]
I'll try to find time for release this weekend.

**Pull request merged**